templates/tfengine/components/cicd/main.tf

```diff
@@ -250,6 +254,41 @@ resource "google_project_iam_member" "cloudbuild_scheduler_sa_project_iam" {
 }
 {{- end}}
 
+{{- if not (get . "service_account_exists")}}
+resource "google_service_account" "cloudbuild_sa" {
+  project      = var.project_id
+  account_id   = var.service_account
+  display_name = "Cloudbuild service account"
+  description  = "Cloudbuild service account"
+}
+{{- end}}
+
+module "logs_bucket" {
+  source  = "terraform-google-modules/cloud-storage/google//modules/simple_bucket"
+  version = "~> 1.4"
+
+  name       = var.logs_bucket
+  project_id = var.project_id
+  location   = "{{get . "storage_location" $.storage_location}}"
+}
+
+module "project_iam_members" {
+  source  = "terraform-google-modules/iam/google//modules/projects_iam"
+  version = "~> 7.2.0"
+
+  projects = [var.project_id]
+  mode     = "additive"
+
+  bindings = {
```

Comment:
These additional bindings should be added to the existing list at the top (in the locals). From looking at https://cloud.google.com/build/docs/securing-builds/configure-user-specified-service-accounts#permissions, `roles/logging.logWriter` is needed to write logs. I think granting it in the `devops` project should be enough, but might need it on the folder level, but we can try on the devops project first. `cloudbuild_devops_roles` is used for this, and we should add the new role there.

Is `roles/iam.serviceAccountUser` needed? The doc says "the user requesting the build" needs this role. I'm not sure why the SA itself need this role, unless i miss anything here.

Reply:
Done.
Since we always create the `logs_bucket` in the devops project then granting the permission at the project level should be enough.